index.ts

```diff
@@ -1,22 +1,35 @@
-import Gun from "gun";
-const program = require('commander');
-
-program
-    .option('-p, --port <number>', 'port number', 3000)
-    .parse(process.argv);
-
-const port = program.port;
+// use import in ts. As fas as I can tell, require does not import the types correctly.
+import { program } from 'commander';
+
+/**
+ * Throw an error if port is not provided.
+ * parse() without passing in "process.argv" is impicit and auto-detects falgs.
+ * to add a default value to port, pass an additional argument as string with the
+ * value at the end.
+ * eg: program.requiredOption('-p, --port <number>', 'port number', '3000).parse();
+ */
+program.requiredOption('-p, --port <number>', 'port number').parse();
+
+/**
+ * parse the provided port as a number
+ */
+const port: number = parseInt(program.opts().port, 10);
```

Comment:
0-1023 are reserved for system level functions. They are classified as well known ports. The default being 3000 is fine.